ibm/resource_ibm_private_dns_zones.go

```diff
@@ -0,0 +1,159 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	iserrors "github.ibm.com/Bluemix/riaas-go-client/errors"
+)
+
+const (
+	pdnsInstanceID      = "instance_id"
+	pdnsZoneName        = "name"
+	pdnsZoneDescription = "description"
+	pdnsZoneLabel       = "label"
+)
+
+func resourceIBMPrivateDNSZone() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMPrivateDnsZoneCreate,
+		Read:     resourceIBMPrivateDnsZoneRead,
+		Update:   resourceIBMPrivateDnsZoneUpdate,
+		Delete:   resourceIBMPrivateDnsZoneDelete,
+		Exists:   resourceIBMPrivateDnsZoneExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		CustomizeDiff: customdiff.Sequence(
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				return resourceTagsCustomizeDiff(diff)
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsZoneName: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsZoneDescription: {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+			},
+
+			pdnsZoneLabel: {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+			},
```

Comment:
We have update support for description and label
https://cloud.ibm.com/docs/dns-svcs?topic=dns-svcs-cli-plugin-dns-services-cli-commands#update-zone